File: test_project/js/example_with_spdx.js
/* SPDX-License-Identifier: Apache-2.0
 *
 * Copyright (c) 2025 Example JavaScript Project
 */

function greetUser(name) {
    console.log(`Hello, ${name}! Welcome to SPDX Scanner.`);
}

class MathUtils {
    static add(a, b) {
        return a + b;
    }

    static multiply(a, b) {
        return a * b;
    }
}

if (typeof module !== 'undefined' && module.exports) {
    module.exports = { greetUser, MathUtils };
}

File: test_project/python/example_with_spdx.py
#!/usr/bin/env python3
"""
SPDX-License-Identifier: MIT

Copyright (c) 2025 Example Corporation
"""

import json
import sys


class ExampleClass:
    """Example class for testing SPDX scanner."""

    def __init__(self, name: str):
        self.name = name

    def get_name(self) -> str:
        """Get the name of the instance."""
        return self.name


def main():
    """Main function."""
    print("Hello, SPDX Scanner!")
    example = ExampleClass("test")
    print(f"Name: {example.get_name()}")


if __name__ == "__main__":
    main()